src/client/app/store/app-entity-store.module.ts

```diff
@@ -1,42 +1,48 @@
 import { NgModule } from '@angular/core';
 import { StoreModule } from '@ngrx/store';
+
 import { EffectsModule } from '@ngrx/effects';
 import {
   EntityCache,
   EntityCollection,
   EntityEffects,
-  EntityDataService,
-  reducer
+  EntityDataServiceConfig,
+  entityReducer,
+  NgrxDataModule,
+  PLURALIZER_NAMES,
 } from '../../ngrx-data';
 
-import { AppDataService, services } from './services';
-import { Hero, Villain } from '../core';
-import { appConfigReducers } from './app-config/reducer';
-import { appConfigServices } from './app-config';
+import { appConfigReducers, appConfigServices } from './app-config';
 
 // This has to be an object and not a new() due to AOT
-const initialEntityCollectionState: EntityCollection<any> = {
-  filter: '',
-  entities: [],
-  filteredEntities: [],
-  loading: false
-};
+const initialEntityCollectionState = new EntityCollection();
```

Comment:
Our `TODO` reminds us that this cannot be a function as AOT won't complete successfully.

Reply:
Should be fine. It's not used in metadata. I came up with different way.